File: examples/python/usage.py
from pathlib import Path

from blackroad_core import Catalog, RoleGuard

catalog = Catalog.load(str(Path(__file__).parent.parent / "tests" / "fixtures" / "catalog.json"))
guard = RoleGuard(["admin"], policy_path=str(Path(__file__).parent.parent / "policy" / "role_matrix.yaml"))
print("Agents", len(catalog.agents))
print("Can manage policy?", guard.can_perform("manage", "policy"))

File: examples/node/usage.ts
import { loadCatalog, RoleGuard } from "@blackroad/core";

async function main() {
  const catalog = await loadCatalog("../tests/fixtures/catalog.json");
  const guard = new RoleGuard(["operator"]);
  console.log("Agents", catalog.agents.length);
  console.log("Can execute?", guard.canPerform("execute", "task"));
}

main();
